ssh.go

```diff
@@ -16,18 +16,21 @@ import (
 
 // Client is a wrapper over the SSH connection/sessions.
 type SSHClient struct {
-	conn         *ssh.Client
-	sess         *ssh.Session
-	user         string
-	host         string
-	remoteStdin  io.WriteCloser
-	remoteStdout io.Reader
-	remoteStderr io.Reader
-	connOpened   bool
-	sessOpened   bool
-	running      bool
-	env          string //export FOO="bar"; export BAR="baz";
-	color        string
+	conn               *ssh.Client
+	sess               *ssh.Session
+	user               string
+	host               string
+	sshKeys            []string // ssh key to use
+	remoteStdin        io.WriteCloser
+	remoteStdout       io.Reader
+	remoteStderr       io.Reader
+	connOpened         bool
+	sessOpened         bool
+	running            bool
+	env                string //export FOO="bar"; export BAR="baz";
+	color              string
+	initAuthMethodOnce sync.Once
```

Comment:
Makes perfect sense, I'll move it back up a level in scope to preserve that functionality.